### ISSUE TITLE
Japanese string comes out garbled in LS 5.6.0 or higher 

### ISSUE THREAD

**Issue by TomonoriSoejima** [elastic/logstash#9708]
It appears that in LS 5.6.0 or higher when you use Japanese in the field name, they come out garbled.
The behavior was not seen in 5.5.x.
I reproduced this behavior in 

ES 6.2.2 
LS 6.2.2

```
cat /path/to/sample2.csv
first,last
tom,cruise
nick,white
```

```
input {
  file {
    path => "/path/to/sample2.csv"
    start_position => "beginning"
    sincedb_path => "/dev/null"
  }
}

filter {
  csv {
      separator => ","
      skip_header => true
      columns => ["テストフィールド  ","last"]
  }

  if [first] == "tom" {
      mutate {add_tag => [ "tom_found"] }
  }
    
}


output {
  stdout { codec => rubydebug }
    elasticsearch {
      hosts => "localhost:9200"
      index => "normal_index"
  }
 }

``` 

search result.

```
{
  "took": 7,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 3,
    "max_score": 1,
    "hits": [
      {
        "_index": "normal_index",
        "_type": "doc",
        "_id": "4MVy02MBXkfuJK1mGBNa",
        "_score": 1,
        "_source": {
          "@timestamp": "2018-06-06T04:54:19.434Z",
          "message": "first,last",
          "path": "/Users/surfer/elastic/labs/logstash/logstash.config/sample2.csv",
          "ãã¹ããã£ã¼ã«ã  ": "first",
          "@version": "1",
          "host": "wave.local",
          "last": "last"
        }
      },
      {
        "_index": "normal_index",
        "_type": "doc",
        "_id": "4cVy02MBXkfuJK1mGRMP",
        "_score": 1,
        "_source": {
          "@timestamp": "2018-06-06T04:54:19.516Z",
          "message": "nick,white",
          "path": "/Users/surfer/elastic/labs/logstash/logstash.config/sample2.csv",
          "ãã¹ããã£ã¼ã«ã  ": "nick",
          "@version": "1",
          "host": "wave.local",
          "last": "white"
        }
      },
      {
        "_index": "normal_index",
        "_type": "doc",
        "_id": "38Vy02MBXkfuJK1mGBNO",
        "_score": 1,
        "_source": {
          "@timestamp": "2018-06-06T04:54:19.514Z",
          "message": "tom,cruise",
          "path": "/Users/surfer/elastic/labs/logstash/logstash.config/sample2.csv",
          "ãã¹ããã£ã¼ã«ã  ": "tom",
          "@version": "1",
          "host": "wave.local",
          "last": "cruise"
        }
      }
    ]
  }
}
```

**Comment by TomonoriSoejima** [comment]
After looking further, we found that input plugin can deal with Japanese text correctly.
We verified it by using file input plugin with JSON file which contains Japanese as a field name.
Thus it appears that `filter` is what is having the issue.

**Comment by original-brownbear** [comment]
@TomonoriSoejima are you using the persistent queue when this happens?

**Comment by TomonoriSoejima** [comment]
Good point, let me check that on the customer.

**Comment by TomonoriSoejima** [comment]
@original-brownbear No luck, but found additional insights.

When you look at it with `stdout { codec => rubydebug }`, you can see that the field is turned into UNICODE.

```
{

                                          "host" => "node-1",

                                         "message" => "テスト",

                                        "@version" => "1",

  "a\u0083\u0086a\u00821a\u0083\u0088a\u0083\u0095a\u0082￡a\u0083?a\u0083≪a\u0083\u0089" => "テスト",

                                       "@timestamp" => 2018-06-12T01:15:08.162Z

}

```

But it appears that conversion is failing? 
Becuase you see characters like 
```
a
￡
?
≪
```

Also

 `テ` which should have been
```
\u0083
```

instead of 
```
\u30C6
```

**Comment by jakelandis** [comment]
I was able to reproduce this as an issue introduced with `5.6.0`. Specifically it appears [this commit](https://github.com/elastic/logstash/commit/5aca82f3fff1c5a8835459cfd216a34f70669784) changed the behavior.  

Using the provided example (file/csv + stdout), I was able to fix it by reverting reference.getByteList() back to --> reference.asJavaString().  I am not sure why it changed or if this is the correct fix. 

@original-brownbear - can you take it from here ? 


```patch
diff --git a/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java b/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java
index 6e2ec8548..c254336cf 100644
--- a/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java
+++ b/logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java
@@ -82,14 +82,14 @@ public final class JrubyEventExtLibrary {
         {
             return Rubyfier.deep(
                 context.runtime,
-                this.event.getUnconvertedField(FieldReference.from(reference.getByteList()))
+                this.event.getUnconvertedField(FieldReference.from(reference.asJavaString()))
             );
         }
 
         @JRubyMethod(name = "set", required = 2)
         public IRubyObject ruby_set_field(ThreadContext context, RubyString reference, IRubyObject value)
         {
-            final FieldReference r = FieldReference.from(reference.getByteList());
+            final FieldReference r = FieldReference.from(reference.asJavaString());
             if (r.equals(FieldReference.TIMESTAMP_REFERENCE)) {
                 if (!(value instanceof JrubyTimestampExtLibrary.RubyTimestamp)) {
                     throw context.runtime.newTypeError("wrong argument type " + value.getMetaClass() + " (expected LogStash::Timestamp)");

```

**Comment by original-brownbear** [comment]
@jakelandis thanks, makes sense will fix that :)

**Comment by original-brownbear** [comment]
fix incoming in https://github.com/elastic/logstash/pull/9764

**Comment by original-brownbear** [comment]
fixed in #9764
